### PR TITLE
Check that the intercepted method actually has any cache annotation declared.

### LINKED ISSUE #257: A function in a class with @CacheConfigs gets processed even without @Cacheable or other such annotations
It seems like https://github.com/micronaut-projects/micronaut-cache/blob/80222a6ee95856c9c1fd913b1a7af5adff8a0444/cache-core/src/main/java/io/micronaut/cache/interceptor/CacheInterceptor.java#L118 checks for `CacheConfigs` stereotype, and intercepts functions not annotated at all in the whole class, which leads to bugs like functions returning `Flow` returning only one value (side-effect of https://github.com/micronaut-projects/micronaut-cache/blob/80222a6ee95856c9c1fd913b1a7af5adff8a0444/cache-core/src/main/java/io/micronaut/cache/interceptor/CacheInterceptor.java#L153). Which is wrong when not even trying to save a cache on such a function.

It should not even try to process such a function.

### PR DESCRIPTION
**Pull request by alvarosanchez**
Fixes #257

### CODE REVIEW

**Comment by jameskleeh** [comment]
The changes as made here can break scenarios with inheritance. Instead of relying on `@CacheConfig`, there should be another meta annotation, perhaps called `@CacheOperation`, that each of the annotations use. That new annotation should have the `@Around
@Type(CacheInterceptor.class)` instead of cache config.

**Comment by alvarosanchez** [comment]
@jameskleeh done. Please review again

File: cache-core/src/main/java/io/micronaut/cache/annotation/InvalidateOperations.java [hunk 1]
```diff
@@ -32,6 +32,7 @@
 @Retention(RetentionPolicy.RUNTIME)
 @Inherited
 @Documented
+@CacheAnnotation
```
Comment (jameskleeh) [inline cache-core/src/main/java/io/micronaut/cache/annotation/InvalidateOperations.java:35]: Why was this added here? If `@CacheConfig` was not already there then it shouldn't need to be here either

Comment (alvarosanchez) [reply cache-core/src/main/java/io/micronaut/cache/annotation/InvalidateOperations.java:35]: Actually, it is required. When removed, the TCK fails for tests that hit methods like:

```groovy
@PutOperations([
        @CachePut('counter'),
        @CachePut('counter2')

])
int increment2(String name) {
    int value = counters2.computeIfAbsent(name, { 0 })
    counters2.put(name, ++value)
    return value
}
```

**Pull request merged**